src/test/java/teammates/test/cases/testdriver/TestNgTest.java

```diff
@@ -0,0 +1,110 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map.Entry;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestNgTest extends BaseTestCase {
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {        
+        String testNgXml = FileHelper.readFile("./src/test/testng.xml");
+        HashMap<String, String> testFiles = getTestFiles(testNgXml, "./src/test/java/teammates/test/cases"); // <class name, package name>
+        
+        testFiles = excludeFilesNotInTestNg(testFiles);
+        
+        for (Entry<String, String> testFileName : testFiles.entrySet()) {
+            assertTrue(isTestFileIncluded(testNgXml, testFileName.getValue(), testFileName.getKey()));
+        }
+    }
+    
+    /**
+     * Files to be checked in testng.xml are added to testFiles
+     * 
+     * @param testNgXml    Contents of testng.xml
+     * @param rootPath     Root path of test files
+     * @return             HashMap containing <class name, package name>
+     */
+    private HashMap<String, String> getTestFiles(String testNgXml, String rootPath) {       
+        // BaseComponentTestCase, BaseTestCase (files in current directory) excluded because 
+        // base classes are extended by the actual tests
+        
+        return addFilesToTestsRecursively(rootPath, true, "teammates.test.cases", testNgXml); 
+    }
+    
+    /**
+     * Exclude files which do not have tests in TestNG
+     * 
+     * @param testFiles    Files to be checked before excluding tests
+     * @return             Files to be checked after excluding tests
+     */
+    private HashMap<String, String> excludeFilesNotInTestNg(HashMap<String, String> testFiles) {
+        // Base*TestCase are base classes to be extended by the actual tests
+        testFiles.remove("BaseUiTestCase");
+        
+        // FeedbackQuestionUiTest is the base class for all Feedback*QuestionUiTest (different question types)
+        testFiles.remove("FeedbackQuestionUiTest");
+        
+        // Needs to be run only when changes are made to GodMode
+        testFiles.remove("GodModeTest");
```

Comment:
Perhaps you can define a the list of excluded test cases at the top and feed it to this method? That way, it will be easier to test and also easier to modify the list of exclusions later.